src/features/hotspots/setup/HotspotSetupExternalScreen.tsx

```diff
@@ -0,0 +1,243 @@
+        )
+      }
+    },
+    [],
+  )
+
+  const linkToMaker = useMemo(() => {
+    const { onboardUrl } = HotspotMakerModels[params.hotspotType]
+
+    if (!onboardUrl) return null
+    const url = onboardUrl.replace(/WALLET/, address || '')
+    return (
+      <TouchableOpacity onPress={openMakerUrl(url)}>
+        <Text
+          variant="bold"
+          fontSize={{ smallPhone: 15, phone: 19 }}
+          color="purpleMain"
+          lineHeight={{ smallPhone: 20, phone: 26 }}
+          maxFontSizeMultiplier={1}
+          numberOfLines={1}
+          marginVertical="m"
+        >
+          {url}
+        </Text>
+      </TouchableOpacity>
+    )
+  }, [address, openMakerUrl, params.hotspotType])
+
+  const subtitle = useMemo(() => {
+    const onboard = t(`makerHotspot.${params.hotspotType}.externalOnboard`, {
+      returnObjects: true,
+    })
+
+    const isArray = Array.isArray(onboard)
+    if (isArray && onboard.length) {
+      return onboard[0]
+    }
+    if (!isArray) {
+      return onboard
+    }
+    return ''
+  }, [params.hotspotType, t])
+
+  const additionalPhrases = useMemo(() => {
+    const onboard = t(`makerHotspot.${params.hotspotType}.externalOnboard`, {
+      returnObjects: true,
+    })
+
+    const isArray = Array.isArray(onboard)
+    if (isArray && onboard.length > 1) {
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const [first, ...rest] = onboard
+      return rest
+    }
+  }, [params.hotspotType, t])
+
+  const scrollViewStyle = useMemo(() => ({ borderRadius: xl }), [xl])
+
+  return (
+    <BackScreen
+      backgroundColor="primaryBackground"
+      paddingTop={{ smallPhone: 's', phone: 'lx' }}
+      onClose={handleClose}
+    >
+      <ScrollView showsVerticalScrollIndicator={false} style={scrollViewStyle}>
+        <Box
+          height={52}
+          width={52}
+          backgroundColor="purple500"
+          borderRadius="m"
+          alignItems="center"
+          justifyContent="center"
+        >
+          {isQr ? (
+            <QrIcon color={colors.purpleMain} width={24} height={24} />
+          ) : (
+            <LinkIcon color={colors.purpleMain} width={30} height={30} />
+          )}
+        </Box>
+        <Text
+          variant="h1"
+          numberOfLines={1}
+          lineHeight={{ smallPhone: 42, phone: 62 }}
+          fontSize={{ smallPhone: 28, phone: 40 }}
+          adjustsFontSizeToFit
+          marginTop="s"
+        >
+          {t(`hotspot_setup.external.${isQr ? 'qr' : 'web'}Title`)}
+        </Text>
+        <Text
+          variant="subtitle"
+          fontSize={{ smallPhone: 15, phone: 19 }}
+          lineHeight={{ smallPhone: 20, phone: 26 }}
+          maxFontSizeMultiplier={1}
+          marginTop={{ smallPhone: 's', phone: 'l' }}
+          marginBottom={
+            linkToMaker ? undefined : { smallPhone: 's', phone: 'l' }
+          }
+        >
+          {subtitle}
+        </Text>
+        {linkToMaker}
```

Comment:
might be worth breaking things like this out into proper components since that's really all it is